Entities/HDFirearm.cpp

```diff
@@ -811,7 +822,7 @@ void HDFirearm::Update()
 
                     // Set the fired particle to not hit this HeldDevice's parent, if applicable
                     if (m_FireIgnoresThis)
-                        pParticle->SetWhichMOToNotHit(pRootParent, 1.0f);
+                        pParticle->SetWhichMOToNotHit(this, 1.0f);
```

Comment:
Is this right?
Setting `this` means only the gun itself is ignored, and the actor holding is it not, which would be pretty bad.

Reply:
It seems to translate up to not hitting the actor when they're holding the gun, I've shot through myself fine with it, even with friendly fire turned off via the global script.
This was needed cause 4zK noticed that sometimes when a gun got launched out of your hand cause of strong forces, its bullet would hit it, which isn't great. Longterm solution is change WhichMOToNotHit to a vector or whatever, so you can just specify multiple stuff.